Add support for wiremesh rendering to cross section render engine

HenryWConklin opened the pull request:
Always uses an orthographic projection for 4D -> 3D

#5

aaronfranke reviewed (commented):
Some bugs:

* The `material` property on meshes is not being used, only the MeshInstance4D `material_override` is being used. If `material_override` is null, it should use the `material` from the mesh. There is a helper function `MeshInstance4D::get_active_material()` which gives you the correct material.
* Wireframe meshes without a material are drawn on top of the camera.

HenryWConklin commented:
Ok, fixed the weird behavior with no material set by adding a default material. Also fixed the issue with not using the material on the mesh if the mesh instance override wasn't set.

aaronfranke commented:
Apologies for the slow review, I missed that this was ready and though it was still WIP.

With this PR, I am getting this error only in the project manager, because it uses the compatibility renderer:

```
SHADER ERROR: Uniform instances are not supported in gl_compatibility shaders.
          at: (null) (:6)
ERROR: Shader compilation failed.
   at: set_code (drivers/gles3/storage/material_storage.cpp:2973)
```

I also think we should rename "default material" to "fallback material" since it's clearer that it's only used as a fallback when no material is set.

aaronfranke reviewed (approved):
I pushed some changes to this PR, this is good enough to merge now, and future PRs can improve it further. This is a great improvement though, thank you for adding this!
